stdlib/array.gr

```diff
@@ -92,6 +92,24 @@ export let mapi = (fn, array) => {
   })
 }
 
+export let reduce = (fn, initial, array) => {
+  let acc = box(initial);
+  forEach((el) => acc := fn(unbox(acc), el), array);
+  unbox(acc)
+}
+
+export let flatMap = (fn, array) => {
+  reduce((result, value) => append(result, fn(value)), [>], array)
```

Comment:
I know that this makes the code a bit more complicated, but we should probably use `Array.concat` instead of append since it's a lot more efficient.

Reply:
Can you explain how Array.concat is faster?

Reply:
Chatted over discord.